scripts/unity/func_name_list.py

```diff
@@ -13,6 +13,10 @@ def func_names_from_header(in_file, out_file):
         content = f_in.read()
 
     with open(out_file, 'w') as f_out:
+        # remove attributes
+        attribute_pattern = re.compile(r'__attribute__\(\(.+?(?=(\s|;))', re.M | re.S)
```

Comment:
Why `\s`?

Parsing balanced parenthesis languages with regular expressions has well known problems; I think this could use a few more comments at the very least.

Reply:
added comment, used `\s` for case when attribute is added before declaration. In that case it is followed by space.

Reply:
this matches the following string: `'__attribute__((x;'` do you really want that?